lib/github.ml

```diff
@@ -221,36 +208,39 @@ let run_with_auth ~dry_run auth curl k =
   Sos.run_io ~dry_run curl (OS.Cmd.in_string auth) k
 
 let curl_create_release ~token ~dry_run curl version msg user repo =
-  let parse_release_id resp =
-    (* FIXME this is retired. *)
-    let headers = String.cuts ~sep:"\r\n" resp in
-    try
-      let not_slash c = not (Char.equal '/' c) in
-      let loc = List.find (String.is_prefix ~affix:"Location:") headers in
-      let id = String.take ~rev:true ~sat:not_slash loc in
-      match String.to_int id with
-      | Some id -> Ok id
-      | None ->
-          R.error_msgf "Could not parse id from location header %S: %S" loc id
-    with Not_found ->
-      R.error_msgf "Could not find release id in response:\n%s."
-        (String.concat ~sep:"\n" headers)
+  let parse_release_id json =
+    match Yojson.Basic.Util.member "Location" json with
+    | `Int i -> R.ok i
+    | `String s -> R.ok (int_of_string s)
+    | `Null ->
+        R.error_msgf "Could not find release id in response:\n%a."
+          Yojson.Basic.pp json
+    | field ->
+        R.error_msgf "Could not parse location from header: %S"
+          (Yojson.Basic.to_string field)
+  in
+  let parse_upload_url json =
+    match Yojson.Basic.Util.member "upload_url" json with
+    | `String s -> R.ok s
+    | `Null ->
+        R.error_msgf "Could not find release upload url in response:\n%a."
+          Yojson.Basic.pp json
+    | field ->
+        R.error_msgf "Could not parse upload url from header: %S"
+          (Yojson.Basic.to_string field)
   in
   let data = create_release_json version msg in
   let uri = strf "https://api.github.com/repos/%s/%s/releases" user repo in
   github_auth ~dry_run ~user token >>= fun auth ->
   let cmd = Cmd.(curl % "-D" % "-" % "--data" % data % uri) in
   run_with_auth ~dry_run ~default:"Location: /0" auth cmd
     (OS.Cmd.to_string ~trim:false)
-  >>= parse_release_id
+  >>| Yojson.Basic.from_string
+  >>= fun str ->
+  parse_release_id str >>= fun id ->
+  parse_upload_url str >>= fun url -> R.return (id, url)
 
-let curl_upload_archive ~token ~dry_run curl archive user repo release_id =
-  let uri =
-    (* FIXME upload URI prefix should be taken from release creation
-         response *)
-    strf "https://uploads.github.com/repos/%s/%s/releases/%d/assets?name=%s"
```

Comment:
This is a good fix, could you move it to a separate commit and add an changelog entry for this fix in particular. It can live in the same PR but I think it's worth mentioning it in the changelog!

Reply:
It looks difficult to isolate these changes, as we rely on other functions that would have been modified and the previous commit would look half-done.
Wouldn't it be enough to add more details in CHANGES.md instead?

Reply:
Ok, don't bother with that then. The extra changelog entry will do indeed!

Reply:
changelog modified